### DISCUSSION TITLE
Lexical -> JSON -> HTML?

### DISCUSSION THREAD

**Discussion by jpnws** [facebook/lexical#5091] [Q&A]
I saved lexical content as JSON in my database. Now I want to display this content on a web page. I checked the following webpage: https://lexical.dev/docs/concepts/serialization, but couldn't find a straightforward answer. Let's say I retrieve the JSON version of some content that I saved using Lexical. How can I take this JSON and display it in the format I originally typed it? For example, if I have italicized, emboldened, or turned some text into a heading, how would I render these elements as HTML? I'm using React and Next.js.

**Comment by anteloio** [comment]
You should be able to use the same component as you used in the edit view, just by changing it to readOnly mode. 

You can find more details about the two modes here: https://lexical.dev/docs/concepts/read-only

But in short, all you should need is to change editable to false on your config: 

```js
<LexicalComposer initialConfig={{editable: false}}>
  ...
</LexicalComposer>
```

**Comment by vadimkantorov** [comment]
Maybe it's useful for your usecase too (e.g. if it's for serverside rendering). I made a CLI tool for editorState JSON -> HTML rendering: https://github.com/vadimkantorov/lexical-cli-render

**Reply by pranavmappoli** [reply]
Is there any hosted example available ?

**Reply by vadimkantorov** [reply]
My repo has a GitHub Actions workflow file as a demo. But otherwise - no hosted example
